--- a/rocks/share/lua/5.1/inifile.lua	2019-06-09 13:47:05.437493211 +0200
+++ b/rocks/share/lua/5.1/inifile.lua	2019-06-09 12:55:01.279083236 +0200
@@ -80,7 +80,7 @@
 		end
 
 		-- Key-value pairs
-		local key, value = line:match("^([%w_]+)%s-=%s-(.+)$")
+		local key, value = line:match("^(%S+)%s-=%s*(.+)$")
 		if tonumber(value) then value = tonumber(value) end
 		if value == "true" then value = true end
 		if value == "false" then value = false end
