--- a/rocks/share/lua/5.1/resty/core/base.lua	2019-02-22 08:36:31.000000000 +0100
+++ b/rocks/share/lua/5.1/resty/core/base.lua	2019-04-19 15:55:18.271702597 +0200
@@ -234,24 +234,11 @@
 
 
 do
-    local exdata
+   local getfenv = getfenv
 
-    ok, exdata = pcall(require, "thread.exdata")
-    if ok and exdata then
-        function _M.get_request()
-            local r = exdata()
-            if r ~= nil then
-                return r
-            end
-        end
-
-    else
-        local getfenv = getfenv
-
-        function _M.get_request()
-            return getfenv(0).__ngx_req
-        end
-    end
+   function _M.get_request()
+      return getfenv(0).__ngx_req
+   end
 end
 
 
